### PR TITLE
Add support for autocomplete suggestion for Here API

### PR DESCRIPTION
**Pull request by PopoSensei**
We found that suggestions were not working for HERE-API. Even though the API supports autocomplete functionalities.

### CODE REVIEW

File: src/geocoders/here.js [hunk 1]
```diff
@@ -28,6 +28,36 @@ export default {
       this.getJSON(this.options.geocodeUrl, params, cb, context);
     },
 
+    geocodeSuggestion: function(query, cb, context) {
```
Comment (perliedman) [inline src/geocoders/here.js]: When/where is this method called?

File: src/geocoders/here.js [hunk 2]
```diff
@@ -28,6 +28,36 @@ export default {
...
+      this.getJSON(this.options.geocodeUrl, params, cb, context);
+    },
+
+    suggest: function(query, cb, context) {
+      var _this = this;
+      var params = {
+        query: query,
+        app_id: this.options.app_id,
+        app_code: this.options.app_code,
+        country: this.options.country
```
Comment (simon04) [inline src/geocoders/here.js]: The (optional) country should be set via `options.suggestQueryParams` (cf. `geocodingQueryParams`, `reverseQueryParams`)

File: src/geocoders/here.js [hunk 3]
```diff
@@ -28,6 +28,36 @@ export default {
...
 
+    geocodeSuggestion: function(query, cb, context) {
+      var params = {
+        locationid: query,
+        gen: 9,
+        app_id: this.options.app_id,
+        app_code: this.options.app_code,
+        jsonattributes: 1
+      };
+      params = L.Util.extend(params, this.options.geocodingQueryParams);
```
Comment (simon04) [inline src/geocoders/here.js]: `suggestQueryParams`

File: src/geocoders/here.js [hunk 4]
```diff
@@ -40,6 +70,16 @@ export default {
       params = L.Util.extend(params, this.options.reverseQueryParams);
       this.getJSON(this.options.reverseGeocodeUrl, params, cb, context);
     },
+    _parseResults: function(data) {
```
Comment (simon04) [inline src/geocoders/here.js]: Please use an inline callback.

**Comment by Daredzik** [comment]
What about it ?? this feature is a 'nice to have' :)

**Comment by PopoSensei** [comment]
Hi, Sorry for not submitting any required changes. Been real busy with school and work. So forgot totally that I submitted this merge request. Will try to have time this holiday to implement the requested changes.

**Comment by Daredzik** [comment]
@PopoSensei ping

**Comment by simon04** [comment]
Closing due to inactivity. Feel free to reopen once you can address the issues from my review.

**Pull request closed without merging**